compiler/src/dotty/tools/dotc/core/SymbolLoaders.scala

```diff
@@ -329,6 +335,16 @@ abstract class SymbolLoader extends LazyType { self =>
     def description(using Context): String = "proxy to ${self.description}"
   }
 
+  private var finalizer = emptyFinalizer
+
+  def withFinalizer(finalizer: Finalizer): SymbolLoader =
+    val cpy = clone().asInstanceOf[SymbolLoader]
+    cpy.finalizer = { sym =>
+      this.finalizer(sym)
```

Comment:
should there be a local reference to `this.finalizer` to avoid capturing the whole SymbolLoader if unnecessary?